### ISSUE TITLE
Keras on Mac (M4) is giving inconsistent results compared to running on NVIDIA GPUs

### ISSUE THREAD

**Issue by To3No7** [keras-team/keras#21108]
For a while now (at least a month or two) I have seen inconsistent results for my Colab notebooks running locally on a Mac M4, compared to running the same notebook code on either T4 (in colab) or RTX3090 locally.

**Comment by To3No7** [comment]
To illustrate the problems I have set up a notebook that implements two simple CNN models that solves the Fashion-MNIST problem. [Colab Notebook](https://colab.research.google.com/drive/11BhtHhN079-BWqv9QvvcSD9U4mlVSocB?usp=sharing)
For a good model with 2M parameters I get the following results:
* T4 (Colab, JAX): Test accuracy: 0.925
* 3090 (Local, Jax): Test accuracy: 0.925
* Mac M4 (Local, JAX): Test accuracy: 0.893
* Mac M4 (Local, Tensorflow): Test accuracy: 0.893

That is, I see a significant drop in performance when I run on the Mac M4 compared to the NVIDIA machines, and it seems to be independent of backend. I however do not know how to pinpoint this to either Keras or Apple’s METAL implementation.

Also when I try to save and cross-load models, I again do not get consistency between platforms. Thus I suspect problems both in model.fit and model.evaluate.

**Comment by To3No7** [comment]
My Python libraries:
```
keras                                  3.9.1
tensorflow                             2.19.0
tensorflow-io-gcs-filesystem           0.37.1
tensorflow-metal                       1.2.0
jax                                    0.5.3
jax-metal                              0.1.1
jaxlib                                 0.5.3
```
